Add paulice docs (#5293)
Qiskit/documentation: docs/addons/qiskit-addon-paulice/index.mdx

*   While many stochastic steps in the algorithm are controllable with a random seed, some features have randomness not controllable with a seed. Specifically, the following kwargs values for `add_pauli_checks` will cause undeterministic check picking: `cost="LER"`, `method="genetic"`, and `method="windowed_genetic"`. For deterministic behavior use `add_pauli_checks(..., cost="gamma", method="windowed")`, which are the default values.

### Future work

*   Support for handling non-Clifford systems
*   More support for analyzing postselected noise channel
*   Handling idling noise when picking checks
*   Controllable randomness for logical error rate cost function and genetic search algorithms

## Contributing

The source code is available [on GitHub](https://github.com/Qiskit/qiskit-paulice).

The developer guide is located at [CONTRIBUTING.md](https://github.com/Qiskit/qiskit-paulice/blob/main/CONTRIBUTING.md) in the root of this project’s repository. By participating, you are expected to uphold Qiskit’s [code of conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).

## Citing this package

> If you use this package in your research, use the CITATION.bib \<[https://github.com/Qiskit/qiskit-paulice/blob/main/CITATION.bib](https://github.com/Qiskit/qiskit-paulice/blob/main/CITATION.bib)> file in this project’s repository to cite the appropriate reference(s).

Reviewer: ```suggestion
If you use this package in your research, use the [CITATION.bib](https://github.com/Qiskit/qiskit-paulice/blob/main/CITATION.bib) file in this project’s repository to cite the appropriate reference(s).
```

Although spacetime Pauli checks may be used to implement standalone error detection routines, they are also relevant in the context of error mitigation and error correction. Error detection can complement error mitigation techniques such as probabilistic error cancellation (PEC) by capturing some of the noise affecting gates and measurements, reducing the impact of the noise channel being inverted and thus reducing the sampling overhead. They can also be viewed as an early step toward practical fault tolerance, since implementing stabilizer codes to protect data qubits from logical errors is a core concept of traditional error correction. Since this method provides single-shot access to the quantum state, it can be used in both sampling-based and expectation value-based workflows.

## Getting started

A simple guide to help you get started quickly with this package is available in the [quickstart guide](guides/quickstart).

## Use case examples

> This technique has been used to improve the fidelity of samples from Clifford-dominated circuits up to 50 qubits and 2450 entangling gates [\[1\]](https://arxiv.org/abs/2504.15725).

Reviewer: ```suggestion
This technique has been used to improve the fidelity of samples from Clifford-dominated circuits up to 50 qubits and 2450 entangling gates [\[1\]](#references).
```

## Technical discussion

### Finding good sets of spacetime Pauli checks

> [![Quantum circuit cartoon](/docs/images/addons/qiskit-addon-paulice/paulice.avif)]

Reviewer: ```suggestion
![Quantum circuit cartoon](/docs/images/addons/qiskit-addon-paulice/paulice.avif)
```

The developer guide is located at [CONTRIBUTING.md](https://github.com/Qiskit/qiskit-paulice/blob/main/CONTRIBUTING.md) in the root of this project’s repository. By participating, you are expected to uphold Qiskit’s [code of conduct](https://github.com/Qiskit/qiskit/blob/main/CODE_OF_CONDUCT.md).

## Citing this package

If you use this package in your research, use the CITATION.bib \<[https://github.com/Qiskit/qiskit-paulice/blob/main/CITATION.bib](https://github.com/Qiskit/qiskit-paulice/blob/main/CITATION.bib)> file in this project’s repository to cite the appropriate reference(s).

## License

[Apache License 2.0](https://github.com/Qiskit/qiskit-paulice/blob/main/LICENSE.txt)

## Deprecation policy

We follow [semantic versioning](https://semver.org/). We may occasionally make breaking changes in order to improve the user experience. When possible, we will keep old interfaces and mark them as deprecated, as long as they can co-exist with the new ones. Each substantial improvement, breaking change, or deprecation will be documented in the release notes.

<span id="id3" />

## References

<span id="id4" />

\[[1](#id1)]

> Simon Martiel, Ali Javadi-Abhari, [Low-overhead error detection with spacetime codes](https://arxiv.org/abs/2504.15725), arXiv:2504.15725 \[quant-ph].

Reviewer: ```suggestion
\[[1](#id1)] Simon Martiel, Ali Javadi-Abhari, [Low-overhead error detection with spacetime codes](https://arxiv.org/abs/2504.15725), arXiv:2504.15725 \[quant-ph].
```
